cypress-io/cypress
# fix: do not double handle errors in component testing

- Closes https://github.com/cypress-io/cypress/issues/23920

### User facing changelog


Fix a bug where unhandled exceptions and errors would be logged twice in Component Testing.

### Additional details


Unlike End To End Testing which has two iframes:
- Spec Frame, for the spec.
- AUT Frame, for the user's application.

Component Testing only has one iframe:
- AUT Frame

AUT Frame is _also_ the Spec Frame, since we serve the specs from a dev server - they are bundled as a single file.
We don't want to bind two error handlers, or we end up logging errors twice. For this reason, we conditionally add these event listeners here - Component Testing errors are captured and logged in `contentWindowListeners#bindToListeners` in `cypress/cy.ts`.

### Steps to test


See steps to reproduce in https://github.com/cypress-io/cypress/issues/23920. Make a component that errors when you clic a button, you can see that before we'd log the same error multiple times - now we do not.

Also, you could copy+paste my regression test (added in driver) into develop and watch it fail.

### How has the user experience changed?


### PR Tasks
